RecoParticleFlow/PFSimProducer/plugins/HGCalTrackCollectionProducer.cc

```diff
@@ -0,0 +1,190 @@
+// S. Zenz, 12 February 2015
+//
+// Splits a track collection into two, based on whether they propagate to the HGCal or not
+// Tracks with bad pt resolution (suspected fakes) are dropped and not in either collection
+
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "RecoParticleFlow/PFClusterProducer/interface/InitialClusteringStepBase.h"
+#include "DataFormats/ParticleFlowReco/interface/PFRecHitFraction.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "RecoParticleFlow/PFClusterProducer/interface/PFCPositionCalculatorBase.h"
+#include "RecoParticleFlow/PFTracking/interface/PFTrackAlgoTools.h"
+
+#include "DataFormats/ForwardDetId/interface/ForwardSubdetector.h"
+#include "DataFormats/Common/interface/Handle.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+
+//#include "DataFormats/TrackReco/interface/Track.h"
+#include "DataFormats/ParticleFlowReco/interface/PFRecTrackFwd.h"
+#include "DataFormats/ParticleFlowReco/interface/PFRecTrack.h"
+#include "DataFormats/TrackReco/interface/Track.h"
+
+
+// for track propagation through HGC  
+// N.B. we are only propogating to first layer, so check these later
+#include "DataFormats/TrajectorySeed/interface/PropagationDirection.h"
+#include "TrackingTools/TrajectoryState/interface/TrajectoryStateTransform.h"
+#include "TrackingTools/MaterialEffects/interface/PropagatorWithMaterial.h"
+#include "DataFormats/GeometrySurface/interface/BoundDisk.h"
+#include "Geometry/TrackerGeometryBuilder/interface/TrackerGeometry.h"
+#include "MagneticField/Engine/interface/MagneticField.h"
+#include "Geometry/HGCalGeometry/interface/HGCalGeometry.h"
+
+//geometry records
+#include "Geometry/Records/interface/TrackerDigiGeometryRecord.h"
+#include "MagneticField/Records/interface/IdealMagneticFieldRecord.h"
+#include "Geometry/Records/interface/IdealGeometryRecord.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+
+#include<unordered_map>
+
+class HGCalTrackCollectionProducer : public edm::stream::EDProducer<> {
+
+public:
+  HGCalTrackCollectionProducer( const edm::ParameterSet & );
+private:
+  void produce( edm::Event &, const edm::EventSetup & ) override;
+  void beginLuminosityBlock(const edm::LuminosityBlock&, 
+			    const edm::EventSetup&) override;
+
+  edm::EDGetTokenT<edm::View<reco::PFRecTrack> > src_;
+  
+  // variables needed for copied goodPtResolution function
+  // need to go back and figure out sensible values
+  bool debug_;
+  const std::vector<double> DPtovPtCut_;
+  const std::vector<unsigned> NHitCut_;
+  const bool useIterTracking_;
+  //  const bool _useFirstLayerOnly; // always true now
```

Comment:
is this commented out code needed?
Please remove or add comments inline in the code why the commented out block is relevant